pkg/ipam/poolallocator/allocator.go

```diff
@@ -272,6 +309,63 @@ func (a *IPPoolAllocator) AllocateNext(state v1alpha2.IPAddressPhase, owner v1al
 	return ip, subnetSpec, err
 }
 
+// AllocateReserved allocates the reserved IP if exists, else allocates next
+// available IP. It returns error if pool is exausted, or in case CRD failed to update its state.
+// In case of success, IP pool CRD status is updated with allocated IP/state/resource/container.
```

Comment:
s/IP pool CRD status/IP pool status/

They are specific CRs, not CRD

Reply:
Fixed.